### ISSUE TITLE
Integration reports 0° temperature(s) for stats on initialisation

### ISSUE THREAD

**Issue by washboy** [NigelCoxon/Hass-heatmiser#13]
Many thanks to @NigelCoxon for releasing the "Fix-Deprecated-Constants" branch.

I've only just installed it because I was waiting for an explicit fix for the polling freeze issue(s). I didn't realise that this new branch seems to address error handling too. It appears that, although commands (e.g. set temperature) may not 'take' first time, failure no longer freezes polling or generates UI errors. It's possible to repeat a failed command and have it then succeed. A huge improvement. Thank you, Nigel.

However (there's always a 'but', isn't there?), I've noticed a minor, but annoying, oddity that's been introduced.

__**tl;dr**__: When the integration initialises (after reboot, for example), it immediately reports a 0° temperature for every thermostat before polling them and reporting the actual temperatures (I haven't checked if the same applies to all stat data). This has the effect of injecting a negative 'spike' into the recorded data, resulting in ugly and difficult to read graphs, etc.

I'd be grateful if Nigel could investigate this when next reviewing the code.

Season's Greetings![]()

**Comment by NigelCoxon** [comment]
I haven't made any changes to the error handling (yet). I have noticed for a while that most of the time when I Set a target temperature, I see an error. Sometimes the target temperature has changed , sometimes not.  I will add more error logging to find out what is happening here. I suspect it is something to do with the stat or the RS485 line rather than a bug, but we'll see.

As I mentioned in the Readme for the Fix Deprecated Constants branch, I have changed the initialisation so that an Update is no longer called during setup. This fixes the "taking more than 10 seconds to initialise" error, but it does mean all the stat values are first seen as 0, until the first update is done. You can easily revert the code to its earlier behaviour. 
Edit the file   /homeassistant/custom_components/heatmiser_ndc/climate.py 
and change False to True in line 79 (in the arguments to the add_entities call)
I tried this in Hass itself using the file editor.
If you have lots of stats this may well trigger the error mentioned above.

I could initialise the room temp and target temp to more sensible values, say 20C, which would avoid some of the blips in any  graphs. I might include a configuration option to do the update or not as part of initialisation.

Next step - improve error logging and handling

**Comment by washboy** [comment]
Doh! Sorry Nigel, I should have read the Readme. Lesson learned. Thanks for patiently pointing  me to a solution.

I've now edited my climate.py to reinstate the update during setup and the pesky 0°s are gone again. A usable system once more, yay!

Yes, I also see errors in the log but at least the integration keeps on going now. Odd to think the poll freezing on failure might have been down to the deprecating constants. I wish I were able to understand the code.

Looking forward to any additional fruits of your Winter nights spare time. 

Happy Holidays!
